### ISSUE TITLE
When the keyboard show out, the bottomNavigationBar behave not properly

### ISSUE THREAD

**Issue by tsangpozheng** [flutter/flutter#14310]
## Steps to Reproduce

When the keyboard appears, the bottomNavigationBar in the Scaffold not stay at the bottom and a keyboard height is added to it.

![image](https://user-images.githubusercontent.com/32248/35469528-43737b7e-0371-11e8-9ee2-dbfdd881c8e2.png)

![image](https://user-images.githubusercontent.com/32248/35469531-4ede4b9c-0371-11e8-8a98-2d18dc10687d.png)
![image](https://user-images.githubusercontent.com/32248/35469533-53af0b16-0371-11e8-8fbd-e43486936015.png)

## Flutter Doctor

[✓] Flutter (on Mac OS X 10.13.3 17D47, locale zh-Hans-CN, channel alpha)
    • Flutter at /Users/tsangpo/workspace/sdk/flutter
    • Framework revision 8f65fec5f5 (7 weeks ago), 2017-12-12 09:50:14 -0800
    • Engine revision edaecdc8b8
    • Tools Dart version 1.25.0-dev.11.0
    • Engine Dart version 2.0.0-edge.d8ae797298c3a6cf8dc9f4558707bd2672224d3e

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.0)
    • Android SDK at /Users/tsangpo/workspace/sdk/android-sdk
    • Android NDK at /Users/tsangpo/workspace/sdk/android-sdk/ndk-bundle
    • Platform android-27, build-tools 27.0.0
    • ANDROID_HOME = /Users/tsangpo/workspace/sdk/android-sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[-] iOS toolchain - develop for iOS devices (Xcode 9.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.2, Build version 9C40b
    ✗ libimobiledevice and ideviceinstaller are not installed. To install, run:
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    ✗ ios-deploy not installed. To install:
        brew install ios-deploy
    • CocoaPods version 1.4.0

[✓] Android Studio (version 3.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[✓] IntelliJ IDEA Community Edition (version 2017.3.3)
    • Flutter plugin version 21.0.3
    • Dart plugin version 173.4301.22

[-] Connected devices
    • None

**Comment by ThobyV** [comment]
Please can we see the bottom navigation code?

**Comment by tsangpozheng** [comment]
```
  @override
  Widget build(BuildContext context) {
    return new Scaffold(
      appBar: new AppBar(
        title: new Text('Vultr Client'),
      ),
      body: new Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: <Widget>[
          new Padding(
            padding: new EdgeInsets.all(20.0),
            child: new Text(
              'Please input your API key',
              style: new TextStyle(
                fontSize: 20.0,
                fontWeight: FontWeight.w900,
              ),
            ),
          ),
          const Text('You can find your API key at https://my.vultr.com/\n'
              '1. Account -> User -> Add User -> User API Key\n'
              '2. Account -> API -> Personal Access Token'),
          new Padding(
            padding: new EdgeInsets.only(left: 20.0, right: 20.0, top: 10.0),
            child: new TextField(
              controller: _controller,
              decoration: new InputDecoration(
                hintText: 'API Key',
              ),
            ),
          ),
        ],
      ),

      bottomNavigationBar: new BottomNavigationBar(
        currentIndex: 1,
        onTap: (int index) {
          if(index==0) {
            _controller.clear();
          }else{
            verifyKey(_controller.text);
          }
        },
        items: <BottomNavigationBarItem>[
          new BottomNavigationBarItem(
            icon: new Icon(Icons.clear),
            title: new Text('Clear'),
          ),
          new BottomNavigationBarItem(
            icon: new Icon(Icons.navigate_next),
            title: new Text('Next'),
          ),
        ],
      ),

    );
  }
```

**Comment by Hixie** [comment]
I think this is probably a duplicate of #13475. cc @cbracken who is the assignee of that bug.

Does it occur on master as well?

**Comment by cbracken** [comment]
Closing as dupe of #13475, which was fixed by #14406 (now landed on master).
